src/finput.js

```diff
@@ -284,75 +294,51 @@ class Finput {
    * @param {e} Keyboard event
    */
   onKeydown(e) {
-    const keyInfo = {
-      event: e,
-      code: e.which || e.keyCode,
-      keyName: e.key.toLowerCase(),
+    const currentState = {
       caretStart: this.element.selectionStart,
       caretEnd: this.element.selectionEnd,
-      currentValue: this.element.value,
-      newValue: this.element.value
-    }
-
-    const actionType = this.getActionType(keyInfo.keyName, e);
+      value: this.element.value,
+      valid: true
+    };
+    const keyEvent = {
+      keyName: e.key.toLowerCase(),
+      modifierKey: this.isModifierKeyPressed(e)
+    };
 
-    console.debug(actionType);
+    const actionType = this.getActionType(keyEvent);
+    const handler = this.getHandlerForAction(actionType);
+    const newState = handler(currentState, keyEvent, this.options, this._history);
 
-    switch (actionType) {
-      case ACTION_TYPES.NUMBER:
-        keyHandlers.onNumber(keyInfo, this.options);
-        break;
-      case ACTION_TYPES.DECIMAL:
-        keyHandlers.onDecimal(keyInfo, this.options);
-        break;
-      case ACTION_TYPES.MINUS:
-        keyHandlers.onMinus(keyInfo, this.options);
-        break;
-      case ACTION_TYPES.SHORTCUT:
-        keyHandlers.onShortcut(keyInfo, this.options);
-        break;
-      case ACTION_TYPES.BACKSPACE:
-        keyHandlers.onBackspace(keyInfo, this.options.thousands);
-        break;
-      case ACTION_TYPES.DELETE:
-        keyHandlers.onDelete(keyInfo, this.options.thousands);
-        break;
-      case ACTION_TYPES.UNDO:
-        keyHandlers.onUndo(this, e);
-        return;
-      case ACTION_TYPES.REDO:
-        keyHandlers.onRedo(this, e);
-        return;
-      default:
-        const isModifierKeyPressed = (e) => {
-          const isMac = navigator.platform.toUpperCase().indexOf('MAC') >= 0;
-          return isMac ? e.metaKey : e.ctrlKey;
-        }
-
-        // all printable characters have a key with length of 1 
-        // if a character has got this far it is an invalid character
-        if(e.key.length === 1 && !isModifierKeyPressed(e)){
-          this.options.invalidKeyCallback(this.createInvalidKeyInfo(keyInfo));
-          e.preventDefault();
-        }
-        return;
+    if (!newState.valid) {
+      this.options.invalidKeyCallback(e);
+      e.preventDefault();
+      return;
     }
 
-    const newValue = helpers.partialFormat(keyInfo.newValue, this.options);
-    const currentValue = keyInfo.newValue;
+    const valueWithTDelimiter = helpers.partialFormat(newState.value, this.options);
```

Comment:
These variable names read like typos to me, can they just be changed to their full length?